### PR TITLE
move peak_pipeline into core and rename it as node_pipeline.

### PR DESCRIPTION
**Pull request by samuelgarcia**
Move peak_pipeline into core and rename it as node_pipeline.
Change tests accordingly.

This PR #1948 must be merged first because we need the new generate mechanisim

### CODE REVIEW

File: src/spikeinterface/core/tests/test_node_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
+import pytest
+import numpy as np
+from pathlib import Path
+import shutil
+
+import scipy.signal
```
Comment (alejoe91) [inline src/spikeinterface/core/tests/test_node_pipeline.py:6]: @samuelgarcia you need to add `scipy` in `test_core` extras!

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Thansk @samuelgarcia 

Can you add `scipy` to the `test_core` requirements? That should fix tests!

**Comment by samuelgarcia** [comment]
> Thansk @samuelgarcia
> 
> Can you add `scipy` to the `test_core` requirements? That should fix tests!

I saw, I will remove this scipy fft convolve instead to have a light core test.

File: src/spikeinterface/core/tests/test_node_pipeline.py (hunk 1 above)
Comment (samuelgarcia) [reply src/spikeinterface/core/tests/test_node_pipeline.py]: I removed scipy instead

**Comment by samuelgarcia** [comment]
@alejoe91 this is ready to be reviewed/merged

**Comment by samuelgarcia** [comment]
@alejoe91 : this is mostly a simple move. Could we merge this ?

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**